src/MvvmNano.Forms/MvvmNanoMasterDetailApplication.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.ObjectModel;
+using System.Linq.Expressions;
+
+namespace MvvmNano.Forms
+{
+    /// <summary>
+    /// The entry point of you XF application, if you want to use master detail (fly out menu) navigation.
+    /// </summary>
+    public abstract class MvvmNanoMasterDetailApplication : MvvmNanoApplication 
+    {
+        /// <summary>
+        /// A collection of MasterDetailData that represents available detail pages.
+        /// </summary>
+        public ObservableCollection<MasterDetailData> MasterDetails { get; set; } = new ObservableCollection<MasterDetailData>();
+
+        /// <summary>
+        /// The <see cref="MvvmNanoMasterDetailPage"/> if one is set. 
+        /// </summary>
+        public MvvmNanoMasterDetailPage MasterPage => MainPage as MvvmNanoMasterDetailPage;
+
+        /// <summary>
+        /// Sets up the main page as a <see cref="MvvmNanoMasterDetailPage{TViewModel}"/> for the given view model type.
+        /// </summary> 
+        public new void SetUpMainPage<TViewModel>() where TViewModel : MvvmNanoViewModel
```

Comment:
I chose new instead of override because that makes it possible to cast back to MvvmNanoApplication and use the other implementation, i did this in the demo:

App.cs line 22:
`((MvvmNanoApplication)this).SetUpMainPage<LoginViewModel>();` 

I'm not sure if that is the best solution to start the app with a ContentPage.